src/highlight.cpp

```diff
@@ -42,28 +42,69 @@ namespace g = grammar;
+    [highlight_spec_pager_description]            = L"fish_pager_color_description",
+    [highlight_spec_pager_secondary_background]   = L"fish_pager_color_secondary_background",
+    [highlight_spec_pager_secondary_prefix]       = L"fish_pager_color_secondary_prefix",
+    [highlight_spec_pager_secondary_completion]   = L"fish_pager_color_secondary_completion",
+    [highlight_spec_pager_secondary_description]  = L"fish_pager_color_secondary_description",
+    [highlight_spec_pager_selected_background]    = L"fish_pager_color_selected_background",
+    [highlight_spec_pager_selected_prefix]        = L"fish_pager_color_selected_prefix",
+    [highlight_spec_pager_selected_completion]    = L"fish_pager_color_selected_completion",
+    [highlight_spec_pager_selected_description]   = L"fish_pager_color_selected_description",
 };
 static_assert(VAR_COUNT == HIGHLIGHT_SPEC_MAX, "Every color spec has a corresponding env var");
 
+// Table used to fetch fallback highlights in case the specified one
+// wasn't set.
+static const highlight_spec_t fallbacks[] = {
+    [highlight_spec_normal]                       = highlight_spec_normal,
+    [highlight_spec_error]                        = highlight_spec_normal,
+    [highlight_spec_command]                      = highlight_spec_normal,
+    [highlight_spec_statement_terminator]         = highlight_spec_normal,
+    [highlight_spec_param]                        = highlight_spec_normal,
+    [highlight_spec_comment]                      = highlight_spec_normal,
+    [highlight_spec_match]                        = highlight_spec_normal,
+    [highlight_spec_search_match]                 = highlight_spec_normal,
+    [highlight_spec_operator]                     = highlight_spec_normal,
+    [highlight_spec_escape]                       = highlight_spec_normal,
+    [highlight_spec_quote]                        = highlight_spec_normal,
+    [highlight_spec_redirection]                  = highlight_spec_normal,
+    [highlight_spec_autosuggestion]               = highlight_spec_normal,
+    [highlight_spec_selection]                    = highlight_spec_normal,
+    [highlight_spec_pager_progress]               = highlight_spec_normal,
+    [highlight_spec_pager_background]             = highlight_spec_normal,
+    [highlight_spec_pager_prefix]                 = highlight_spec_normal,
+    [highlight_spec_pager_completion]             = highlight_spec_normal,
+    [highlight_spec_pager_description]            = highlight_spec_normal,
+    [highlight_spec_pager_secondary_background]   = highlight_spec_pager_background,
+    [highlight_spec_pager_secondary_prefix]       = highlight_spec_pager_prefix,
+    [highlight_spec_pager_secondary_completion]   = highlight_spec_pager_completion,
+    [highlight_spec_pager_secondary_description]  = highlight_spec_pager_description,
+    [highlight_spec_pager_selected_background]    = highlight_spec_pager_background,
```

Comment:
These `highlight_spec_pager_selected_*` are fallbacks for the selected row, I think these should fall back to `highlight_spec_selection` otherwise the selected row will not be visible.